plugins/woo-ai/src/shared/productData.ts

```diff
@@ -0,0 +1,128 @@
+/**
+ * Internal dependencies
+ */
+import { Attribute, ProductData, TinyContent } from './types';
+
+declare const tinymce: { get: ( str: string ) => TinyContent };
+
+const isElementVisible = ( element: HTMLElement ) =>
+	! ( window.getComputedStyle( element ).display === 'none' );
+
+const getCategories = (): string[] => {
+	const categoryCheckboxEls: NodeListOf< HTMLInputElement > =
+		document.querySelectorAll(
+			'#taxonomy-product_cat input[name="tax_input[product_cat][]"]:checked'
+		);
+
+	const tempCategories: string[] = [];
+
+	categoryCheckboxEls.forEach( ( el ) => {
```

Comment:
I believe this includes the `Uncategorized` category that is assigned to all by default. We might want to omit that since it isn't useful.

Reply:
I wasn't sure how to do that in the front end since I didn't have the category slug; the ID is generated randomly, and the name can also be variable because of translation. But I added a change in the backend code to omit that category when generating the prompt: https://github.com/woocommerce/woocommerce/pull/38392/commits/7f689a8f86e3db361892d571fa2fcb1bcbb2a58b